enhancements/authentication/oauth-resource-storage.md

```diff
@@ -0,0 +1,197 @@
+}
+
+// OAuthAuthorizeToken describes an OAuth authorization token
+type OAuthAuthorizeToken struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+    ...
+    
+    // SecretToken is the second half of the secret information a user has to pass as bearer token
+    // in order to authorize with the API server. It is concatenated with the name of the 
+    // OAuthAuthorizeToken object.
+    // For old tokens, this field might be empty.
+    // +optional
+    SecretToken string `json:"secretToken,omitempty" protobuf:"bytes,11,opt,name=secretToken"`
+}
+```
+
+### User Stories [optional]
+
+#### Story 1
+
+As an attacker who got access to an etcd backup I do not get access to oauth tokens.
+
+#### Story 2
+
+As a user I can use the API via oc or console to log out other sessions. 
+
+#### Story 3
+
+(optional) As a user I can list my other oauth sessions, including information like client name and expiration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The oauth authorizer in kube-apiserver in v4.6 will split the provided bearer token into 
+the first 43 bytes (256 bits, compare https://github.com/openshift/oauth-server/blob/3539e4065084dc0b5f35e43d7ea49631df99760d/pkg/server/crypto/random.go#L31-L35)
+and remaining bytes. It then looks up the `OAuthAccessToken` via the 
+former as a name (as we do today) and in addition compares the `SecretToken` field with
+the latter (possible zero length).   
+
+Old tokens will have an empty `SecretToken` field, and clients using that token will only
+pass 43 bytes bearer token, leading into the zero length case.
+
+Old kube-apiserver (on downgrade) will try to lookup the token object via the whole token
+and fail. I.e. on downgrade clients have to reauthenticate.
+
+#### Logout and Read Access for Users
+
+With the `SecretToken`, `AuthorizeToken` and `RefreshToken` fields hidden for non-admin users,
+we can allow (via RBAC) normal users to GET, LIST, WATCH, DELETE their own tokens. This can 
+be implemented by injecting a label selector in the List and Watch implementation of the `OAuthAccessToken` 
+registry, similarly as done when proxying project requests to namespaces (where the user
+can list only his own projects; compare https://github.com/openshift/openshift-apiserver/blob/4c982be0e64177b727200e416aa6375f4f17ffc2/pkg/project/apiserver/registry/project/proxy/proxy.go#L83).
+
+For this list by label to work, we have to label new `OAuthAccessToken` with their owner, i.e.
+
+```
+oauth.openshift.io/username: <user-name-matching-the-context-user>
+```
+
+Old tokens won't have this label and won't be visible. As tokens expire after a day (by default; can be customized),
+this is acceptable.
+
+### Risks and Mitigations
+
+- When we add an RBAC role and binding to read OAuthAccessTokens on v4.6 and then downgrade the
```

Comment:
I don't think we should do this.  Creating a new resource == ok, but more work and not needed to resolve the backup problem.  Changing the meaning of an existing resource is a bad idea.

Reply:
Moved this topic out of the release an for not out-of-scope of the enhancement.